src/main/java/it/gov/pagopa/pu/migration/wf/service/ingestion/debtposition/DebtPositionProcessingService.java

```diff
@@ -0,0 +1,162 @@
+  }
+
+  @Override
+  protected boolean consumeRow(long lineNumber, InstallmentIngestionFlowFileDTO dto, DebtPositionMigrationFileResult migrationFileResult, List<DebtPositionErrorDTO> errorList, Uploads uploads) {
+    return true;
+  }
+
+  @Override
+  protected void setNumTotalRows(DebtPositionMigrationFileResult result, long numTotalRows) {
+    result.setNumTotalRows(numTotalRows);
+  }
+
+  @Override
+  protected void setNumCorrectlyProcessedRows(DebtPositionMigrationFileResult result, long numCorrectlyProcessedRows) {
+    result.setNumCorrectlyProcessedRows(numCorrectlyProcessedRows);
+  }
+
+  @Override
+  protected void setErrorDescription(DebtPositionMigrationFileResult result, String errorDescription) {
+    result.setErrorDescription(errorDescription);
+  }
+
+  @Override
+  protected void setDiscardedFileName(DebtPositionMigrationFileResult result, String discardedFileName) {
+    result.setDiscardedFileName(discardedFileName);
+  }
+
+
+
+  /**
+   * Reads and parses the given CSV files, applies default values to each DTO, and collects errors during parsing.
+   * Returns a DebtPositionMigrationFileResult containing the processed files and statistics.
+   *
+   * @param retrievedFiles the list of input CSV files to process
+   * @param errorList the list to populate with DebtPositionErrorDTO errors
+   * @return DebtPositionMigrationFileResult with processed files and stats
+   */
+  public DebtPositionMigrationFileResult readAndParseRows(List<Path> retrievedFiles, List<DebtPositionErrorDTO> errorList) {
+    int numTotalFiles = 0;
+    int numCorrectlyProcessedFiles = 0;
+    List<Path> parsedFiles = new ArrayList<>();
+    List<String> unsuccessfulParsedFiles = new ArrayList<>();
+
+    for (Path file : retrievedFiles) {
+      processSingleFile(file, errorList, parsedFiles, unsuccessfulParsedFiles);
+      numTotalFiles++;
+    }
+
+    numCorrectlyProcessedFiles = parsedFiles.size();
+
+    return DebtPositionMigrationFileResult.builder()
+      .parsedFiles(parsedFiles)
+      .numCorrectlyProcessedFiles(numCorrectlyProcessedFiles)
+      .numTotalFiles(numTotalFiles)
+      .errorDescription(buildErrorDescription(unsuccessfulParsedFiles))
+      .build();
+  }
+
+  private void processSingleFile(Path file, List<DebtPositionErrorDTO> errorList, List<Path> parsedFiles, List<String> unsuccessfulParsedFiles) {
+    try {
+      List<InstallmentIngestionFlowFileDTO> dtos = parseCsvFile(file, errorList, unsuccessfulParsedFiles);
+      // Usa la stessa logica di handleFilesUpload per il nome del file zip temporaneo
+      String originalFileName = file.getFileName().toString();
+      String tempFileName = originalFileName.replaceFirst("\\.[^.]+$", "") + "-parsed.csv";
+      Path tempFile = file.getParent() != null ? file.getParent().resolve(tempFileName) : Path.of(System.getProperty("java.io.tmpdir")).resolve(tempFileName);
+      csvService.createCsv(tempFile, InstallmentIngestionFlowFileDTO.class, () -> dtos, "V2_0");
```

Comment:
this is an infinite loop!
as described on the javadoc, createCsv will call the csvRowsSupplier parameters until it returns an empty collection!
you are always returning the same collection